specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/applicationGateway.json

```diff
@@ -2076,6 +2430,148 @@
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private link resource."
+        }
+      },
+      "description": "Properties of Private Link Resource of an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkResource": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkResourcePropertiesFormat",
+          "description": "Properties of the application gateway private link resource."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the private link resource that is unique within an Application Gateway."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "PrivateLink Resource configuration of an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationGatewayPrivateLinkResource"
+          },
+          "description": "List of private link resources of an application gateway."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of results."
+        }
+      },
+      "description": "Response for ListApplicationGatewayPrivateLinkResources API service call. Gets all private link resources for an application gateway."
+    },
+    "ApplicationGatewayPrivateEndpointConnectionPropertiesFormat": {
+      "properties": {
+        "privateEndpoint": {
+          "readOnly": true,
+          "$ref": "./privateEndpoint.json#/definitions/PrivateEndpoint",
+          "description": "The resource of private end point."
+        },
+        "privateLinkServiceConnectionState": {
+          "$ref": "./privateLinkService.json#/definitions/PrivateLinkServiceConnectionState",
+          "description": "A collection of information about the state of the connection between service consumer and provider."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private endpoint connection resource."
+        },
+        "linkIdentifier": {
```

Comment:
>"linkIdentifier" [](start = 8, length = 16)

like myronFanQui mentioned this deviates from the NRP spec.  If this should be part of the contract everyone should be implementing it, not just applicationGateway.  Private links is supposed to be a consistent contract across RPs that implement it

Reply:
link identifier is common with PrivateLinkService.json.
/privatelinkservice/privateendpointconnection

We have followed that we think we will need this in future. Application gateway uses privatelinkService underneath to provide the privatelink capabilities. Its a read only property for better debugging when supported for customers to figure out where the packet is coming from.